lib/scrabble.rb

```diff
@@ -12,7 +7,30 @@ def point_values
       "M"=>3, "N"=>1, "O"=>1, "P"=>3,
       "Q"=>10, "R"=>1, "S"=>1, "T"=>1,
       "U"=>1, "V"=>4, "W"=>4, "X"=>8,
-      "Y"=>4, "Z"=>10
+      "Y"=>4, "Z"=>10,
     }
   end
+
+  def score(word)
+    if word.nil? || word.empty?
+      0
+    else
+      words = word.chars
+      letter_scores = words.map do |letter|
+      point_values[letter.upcase]
+      end
+      letter_scores.inject { |sum, n| sum + n }
+    end
+  end
+
+  def score_with_multipliers(word, letter_multiplier, word_multiplier)
```

Comment:
This is a bit too long, Look for ways to optimize instead of setting individual steps to local variables.